File: jdroid-java-http-core/src/main/java/com/jdroid/java/http/exception/ConnectionException.java
package com.jdroid.java.http.exception;


import com.jdroid.java.exception.ErrorCodeException;

/**
 * Exception related with connectivity errors.
 */
public class ConnectionException extends ErrorCodeException {
	
	private static final long serialVersionUID = 7201698090980820539L;
	
	private boolean readTimeout = false;
	
	public ConnectionException(Throwable throwable) {
		super(HttpErrorCode.CONNECTION_ERROR, throwable);
	}
	
	public ConnectionException(String message) {
		super(HttpErrorCode.CONNECTION_ERROR, message);
	}
	
	public ConnectionException(Throwable throwable, boolean readTimeout) {
		this(throwable);
		this.readTimeout = readTimeout;
	}
	
	public boolean isReadTimeout() {
		return readTimeout;
	}
}

File: jdroid-java-http-core/src/main/java/com/jdroid/java/http/exception/ConnectionException.kt
package com.jdroid.java.http.exception

import com.jdroid.java.exception.ErrorCodeException

/**
 * Exception related with connectivity errors.
 */
class ConnectionException : ErrorCodeException {

    var isReadTimeout = false

    constructor(throwable: Throwable) : super(HttpErrorCode.CONNECTION_ERROR, throwable) {}

    constructor(message: String) : super(HttpErrorCode.CONNECTION_ERROR, message) {}

    constructor(throwable: Throwable, readTimeout: Boolean) : this(throwable) {
        this.isReadTimeout = readTimeout
    }

    companion object {

        private const val serialVersionUID = 7201698090980820539L
    }
}
